eservice/pdo/eservice/enclave/enclave/block_store.cpp

```diff
@@ -0,0 +1,187 @@
+/* Copyright 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdarg.h>
+#include <stdio.h>
+#include <string.h>
+#include <pthread.h>
+#include <bits/stdc++.h>
+
+#include "pdo_error.h"
+#include "error.h"
+#include "log.h"
+#include "types.h"
+#include "hex_string.h"
+
+#include "enclave/base.h"
+#include "enclave/block_store.h"
+
+static std::unordered_map<std::string, std::string> map;
+static pthread_spinlock_t lock;
+
+pdo_err_t pdo::enclave_api::block_store::BlockStoreInit() {
+    int ret;
+
+    ret = pthread_spin_init(&lock, PTHREAD_PROCESS_SHARED);
+    if (ret != 0) {
+        Log(PDO_LOG_DEBUG, "Failed to init block store spinlock: %d", ret);
+        return PDO_ERR_SYSTEM;
+    }
+
+    return PDO_SUCCESS;
+}
+
+int pdo::enclave_api::block_store::BlockStoreGet(
+    const uint8_t* key,
+    const size_t keySize,
+    uint8_t **value,
+    size_t* valueSize
+    )
+{
+    int result = 0;
+    std::string keyStr = BinaryToHexString(key, keySize);
+    Log(PDO_LOG_DEBUG, "Block Store Get: '%s'", keyStr.c_str());
+
+    // **********
+    // LOCK
+    pthread_spin_lock(&lock);
+
```

Comment:
this is probably best moved into a module that manages the actual storage. That abstraction will simplify any move to persistent backup. and maybe the storage abstraction should be pushed into common?

Reply:
I added a TODO for this - I think it's out of scope of the current commit. We should definitely replace the implementation of this - but I think adding yet another abstraction layer only to stuff this lame implementation in it is not worth doing.